### PR TITLE
Add more protobuf tests

### PR DESCRIPTION
**Pull request by pepone**
This PR adds additional tests for Protobuf support.

### CODE REVIEW

File: src/IceRpc.Protobuf/PipeReaderExtensions.cs [hunk 1]
```diff
@@ -109,7 +109,6 @@ public static class PipeReaderExtensions
 
         if (readResult.Buffer.FirstSpan[0] == 1)
         {
-            reader.AdvanceTo(readResult.Buffer.GetPosition(5));
```
Comment (pepone) [inline src/IceRpc.Protobuf/PipeReaderExtensions.cs:112]: This call to `AdvanceTo` was not required as the caller completes the reader upon exception.

File: tests/IceRpc.Protobuf.Tests/OperationTests.cs [hunk 2]
```diff
@@ -11,14 +11,24 @@ namespace IceRpc.Protobuf.Tests;
 public partial class OperationTests
```
Comment (pepone) [inline tests/IceRpc.Protobuf.Tests/OperationTests.cs:11]: Updated the tests in this file to use nonempty Protobuf messages.

File: tests/IceRpc.Tests.Common/ColocInvoker.cs [hunk 3]
```diff
@@ -66,7 +66,7 @@ public async Task<IncomingResponse> InvokeAsync(OutgoingRequest request, Cancell
             }
             else
             {
-                var pipe = new Pipe();
+                var pipe = new Pipe(new PipeOptions(pauseWriterThreshold: 0));
```
Comment (pepone) [inline tests/IceRpc.Tests.Common/ColocInvoker.cs:69]: This prevents the test from hanging. If we send a large payload with the ColocInvoker, the default pause writer threshold is 16k.

File: tools/IceRpc.ProtocGen/MessageDescriptorExtensions.cs [hunk 4]
```diff
@@ -6,14 +6,11 @@ namespace IceRpc.ProtocGen;
 
 internal static class MessageDescriptorExtensions
 {
-    internal static string GetFullyQualifiedType(this MessageDescriptor messageDescriptor) =>
-        $"{messageDescriptor.File.GetCsharpNamespace()}.{messageDescriptor.Name}";
-
```
Comment (pepone) [inline tools/IceRpc.ProtocGen/MessageDescriptorExtensions.cs:11]: This internal method was no longer used.

**Review by externl — APPROVED** [review]
Looks good!

File: tests/IceRpc.Protobuf.Tests/OperationTests.cs [hunk 5]
```diff
@@ -11,14 +11,24 @@ namespace IceRpc.Protobuf.Tests;
...
-    public void Unary_rpc()
+    public async Task Unary_rpc()
     {
         // Arrange
         var invoker = new ColocInvoker(new MyOperationsService());
         var client = new MyOperationsClient(invoker);
 
-        // Act/Assert
-        Assert.That(async () => await client.UnaryOpWithEmptyParamAndReturnAsync(new Empty()), Throws.Nothing);
+        var m = new InputMessage()
```
Comment (ReeceHumphreys) [inline tests/IceRpc.Protobuf.Tests/OperationTests.cs:20]: I dont think we should use single letter variable names even in tests.

```suggestion
        var message = new InputMessage()
```

File: tests/IceRpc.Protobuf.Tests/OperationTests.cs [hunk 6]
```diff
@@ -11,14 +11,24 @@ namespace IceRpc.Protobuf.Tests;
...
+        {
+            P1 = "P1",
+            P2 = 2,
+        };
+
+        // Act
+        var r = await client.UnaryOpAsync(m);
```
Comment (ReeceHumphreys) [inline tests/IceRpc.Protobuf.Tests/OperationTests.cs:27]:
```suggestion
        var response = await client.UnaryOpAsync(message);
```

File: tests/IceRpc.Protobuf.Tests/OperationTests.cs [hunk 7]
```diff
@@ -11,14 +11,24 @@ namespace IceRpc.Protobuf.Tests;
...
+
+        // Assert
+        Assert.That(r.P1, Is.EqualTo(m.P1));
+        Assert.That(r.P2, Is.EqualTo(m.P2));
```
Comment (ReeceHumphreys) [inline tests/IceRpc.Protobuf.Tests/OperationTests.cs:31]:
```suggestion
        Assert.That(response.P1, Is.EqualTo(message.P1));
        Assert.That(response.P1.P2, Is.EqualTo(message.P1.P2));
```

**Review by ReeceHumphreys — APPROVED** [review]
Looks good.

**Pull request merged**